libs/server/Objects/SortedSet/SortedSetObjectImpl.cs

```diff
@@ -30,37 +30,41 @@ private struct ZRangeOptions
             public bool WithScores { get; set; }
         };
 
-        private void SortedSetAdd(ref ObjectInput input, byte* output)
+        private void SortedSetAdd(ref ObjectInput input, ref SpanByteAndMemory output)
         {
-            var _output = (ObjectOutputHeader*)output;
-
-            var count = input.arg1;
-            *_output = default;
+            var isMemory = false;
+            MemoryHandle ptrHandle = default;
+            var ptr = output.SpanByte.ToPointer();
 
-            var input_startptr = input.payload.ptr;
-            var input_currptr = input_startptr;
-            var length = input.payload.length;
-            var input_endptr = input_startptr + length;
+            var curr = ptr;
+            var end = curr + output.Length;
 
-            for (var c = 0; c < count; c++)
+            ObjectOutputHeader outputHeader = default;
+            var added = 0;
+            try
             {
-                if (!RespReadUtils.ReadDoubleWithLengthHeader(out var score, out var parsed, ref input_currptr, input_endptr))
-                    return;
-                if (!RespReadUtils.TrySliceWithLengthHeader(out var member, ref input_currptr, input_endptr))
-                    return;
-
-                if (parsed)
+                for (var currIdx = input.parseStateStartIdx; currIdx < input.parseState.Count; currIdx += 2)
                 {
-                    var memberArray = member.ToArray();
+                    if (!input.parseState.TryGetDouble(currIdx, out var score))
+                    {
+                        while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_NOT_VALID_FLOAT, ref curr, end))
+                            ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr,
+                                ref end);
```

Comment:
it's a little odd to have the last of several params drop to the next line when there's room on the same line...